index.js

```diff
@@ -44,149 +44,162 @@ module.exports = (UserProps) => {
   return {
     postcssPlugin: 'postcss-jit-props',
 
-    async Once (node, {Rule, AtRule}) {
-      let target_selector = ':root'
+    prepare() {
+      const UserPropsCopy = JSON.parse(JSON.stringify(UserProps));
 
-      if (!Object.keys(UserProps).length) {
-        return console.warn('postcss-jit-props: Variable source(s) not passed.')
-      }
-
-      if (UserProps?.files?.length) {
-
-        const files = UserProps?.files
-          .map((file) => glob(file))
-          .reduce((flattenedFileList, files) => flattenedFileList.concat(files), [])
+      return {
+        async Once(node, { result, Rule, AtRule }) {
+          let target_selector = ':root'
 
-        await Promise.all(files.map(async file => {
-          let data = fs.readFileSync(file, 'utf8')
-          let result = await postcss([(function(){})]).process(data, { from: file })
+          if (!Object.keys(UserPropsCopy).length) {
+            return console.warn('postcss-jit-props: Variable source(s) not passed.')
+          }
 
-          result.root.walkDecls(decl => {
-            if (!decl.prop.includes('--')) return
-            UserProps[decl.prop] = decl.value
-          })
+          if (UserPropsCopy?.files?.length) {
+
+            const files = UserPropsCopy?.files
+              .map((file) => glob(file))
+              .reduce((flattenedFileList, files) => flattenedFileList.concat(files), [])
+
+            await Promise.all(files.map(async file => {
+              result.messages.push({
+                type: 'dependency',
+                plugin: 'postcss-jit-props',
+                file: file,
+                parent: node.source?.input?.file,
+              });
```

Comment:
This is what triggers file watching in tools like PostCSS CLI.

Reply:
rad ty